tensorflow/compiler/tf2xla/kernels/tensor_array_ops.cc

```diff
@@ -311,6 +311,30 @@ class TensorArrayGatherOp : public XlaOpKernel {
 
     xla::ComputationDataHandle ta = resource->value;
 
+    std::vector<int64> const_indices;
+    Status status = ctx->ConstantInputAsIntVector(1, &const_indices);
+    if (status.ok()) {
+      bool is_simple_gather = true;
```

Comment:
This would be more readable if you added a comment defining what a "simple" gather is.

Perhaps use a more descriptive name, maybe "gather_is_dense_slice"?